classes/class-p4-post.php

```diff
@@ -156,28 +164,76 @@ public function get_author_override() {
 		}
 
 		/**
-		 * Get post/page custom planet4 type.
+		 * Gets the tags for this P4_Post.
+		 */
+		public function set_tags() {
+			$tags = wp_get_post_tags( $this->id );
+
+			if ( $tags && ! is_wp_error( $tags ) ) {
+				foreach ( $tags as $tag ) {
+					$this->tags[] = [
+						'name' => $tag->name,
+						'href' => get_tag_link( $tag->term_id ),
+					];
+				}
+			}
+		}
+
+		/**
+		 * Gets the tags for this P4_Post.
+		 */
+		public function get_tags() {
+			$this->tags;
+		}
+
+		/**
+		 * Sets the page type for this P4_Post.
+		 */
+		public function set_page_types() {
+			$page_type_data = get_the_terms( $this->id, P4_Custom_Taxonomy::TAXONOMY );
```

Comment:
Maybe you can use timber's terms() that wrap the objects with Timber\Term and you get the term link also.

Reply:
@kirdia I know, but I did it this way so that it is consistent with the pre-existing code [here](https://github.com/greenpeace/planet4-plugin-blocks/blob/941d4c69b97a349e6083fbd5951307d4f335945d/classes/controller/blocks/class-articles-controller.php#L357) and the way that the [template](https://github.com/greenpeace/planet4-plugin-blocks/pull/393/files#diff-4356b2eb546b563b92942ed0b711c6ddR30) expects the data to render.

Reply:
If I change it, then I will need to change also the populate_post_items() method in the controller.

Reply:
We are in the end of the release. Please open a jira issue for the follow up. IF this works at the moment.